ktlint 0.47.0 throwing error when using rules with prefix ij_kotlin

## Expected Behavior
I updated ktlint from version 0.46.1 to 0.47.0 and when using rules with prefix `ij_kotlin` it throws errors.

Here is my `.editorconfig` file.
```
# http://editorconfig.org
root = true

[*.{kt,kts}]
disabled_rules = filename

ij_kotlin_allow_trailing_comma = true
ij_kotlin_allow_trailing_comma_on_call_site = true
```

With the previous versions it was working.

I have updated ktlint from version 0.46.1 to 0.47.0 and it started throwing the following exception.
If I downgrade ktlint from version 0.47.0 to 0.46.1 it works again.

```
./gradlew ktlintFormat --daemon
Configuration on demand is an incubating feature.

> Task :androidprojectliteandroid:ktlintFormat FAILED
Exception in thread "main" java.util.concurrent.ExecutionException: java.lang.ClassCastException: class java.lang.String cannot be cast to class java.lang.Boolean (java.lang.String and java.lang.Boolean are in module java.base of loader 'bootstrap')
        at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
        at com.pinterest.ktlint.internal.KtlintCommandLine.parallel(KtlintCommandLine.kt:605)
        at com.pinterest.ktlint.internal.KtlintCommandLine.parallel$default(KtlintCommandLine.kt:563)
        at com.pinterest.ktlint.internal.KtlintCommandLine.lintFiles(KtlintCommandLine.kt:321)
        at com.pinterest.ktlint.internal.KtlintCommandLine.run(KtlintCommandLine.kt:268)
        at com.pinterest.ktlint.Main.main(Main.kt:27)
Caused by: java.lang.ClassCastException: class java.lang.String cannot be cast to class java.lang.Boolean (java.lang.String and java.lang.Boolean are in module java.base of loader 'bootstrap')
        at com.pinterest.ktlint.ruleset.standard.TrailingCommaOnCallSiteRule.beforeFirstNode(TrailingCommaOnCallSiteRule.kt:56)
        at com.pinterest.ktlint.core.KtLint.executeRule(KtLint.kt:194)
        at com.pinterest.ktlint.core.KtLint.access$executeRule(KtLint.kt:30)
        at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:252)
        at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:251)
        at com.pinterest.ktlint.core.internal.VisitorProvider$visitor$4.invoke(VisitorProvider.kt:82)
        at com.pinterest.ktlint.core.internal.VisitorProvider$visitor$4.invoke(VisitorProvider.kt:80)
        at com.pinterest.ktlint.core.KtLint.format(KtLint.kt:251)
        at com.pinterest.ktlint.internal.FileUtilsKt.formatFile(FileUtils.kt:254)
        at com.pinterest.ktlint.internal.KtlintCommandLine.process(KtlintCommandLine.kt:379)
        at com.pinterest.ktlint.internal.KtlintCommandLine.access$process(KtlintCommandLine.kt:52)
        at com.pinterest.ktlint.internal.KtlintCommandLine$lintFiles$3.invoke$lambda-0(KtlintCommandLine.kt:312)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)

> Task :androidprojectandroid:ktlintFormat
Exception in thread "main" java.util.concurrent.ExecutionException: java.lang.ClassCastException: class java.lang.String cannot be cast to class java.lang.Boolean (java.lang.String and java.lang.Boolean are in module java.base of loader 'bootstrap')
        at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
        at com.pinterest.ktlint.internal.KtlintCommandLine.parallel(KtlintCommandLine.kt:605)
        at com.pinterest.ktlint.internal.KtlintCommandLine.parallel$default(KtlintCommandLine.kt:563)
        at com.pinterest.ktlint.internal.KtlintCommandLine.lintFiles(KtlintCommandLine.kt:321)
        at com.pinterest.ktlint.internal.KtlintCommandLine.run(KtlintCommandLine.kt:268)
        at com.pinterest.ktlint.Main.main(Main.kt:27)
Caused by: java.lang.ClassCastException: class java.lang.String cannot be cast to class java.lang.Boolean (java.lang.String and java.lang.Boolean are in module java.base of loader 'bootstrap')
        at com.pinterest.ktlint.ruleset.standard.TrailingCommaOnCallSiteRule.beforeFirstNode(TrailingCommaOnCallSiteRule.kt:56)
        at com.pinterest.ktlint.core.KtLint.executeRule(KtLint.kt:194)
        at com.pinterest.ktlint.core.KtLint.access$executeRule(KtLint.kt:30)
        at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:252)
        at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:251)
        at com.pinterest.ktlint.core.internal.VisitorProvider$visitor$4.invoke(VisitorProvider.kt:82)
        at com.pinterest.ktlint.core.internal.VisitorProvider$visitor$4.invoke(VisitorProvider.kt:80)
        at com.pinterest.ktlint.core.KtLint.format(KtLint.kt:251)
        at com.pinterest.ktlint.internal.FileUtilsKt.formatFile(FileUtils.kt:254)
        at com.pinterest.ktlint.internal.KtlintCommandLine.process(KtlintCommandLine.kt:379)
        at com.pinterest.ktlint.internal.KtlintCommandLine.access$process(KtlintCommandLine.kt:52)
        at com.pinterest.ktlint.internal.KtlintCommandLine$lintFiles$3.invoke$lambda-0(KtlintCommandLine.kt:312)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)

> Task :androidprojectandroid:ktlintFormat FAILED

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':androidprojectliteandroid:ktlintFormat'.
> Process 'command '/Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':androidprojectandroid:ktlintFormat'.
> Process 'command '/Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 7s
2 actionable tasks: 2 executed
```

## Observed Behavior
It throws the following exception:

```
./gradlew ktlintFormat --daemon          
Configuration on demand is an incubating feature.

> Task :androidprojectandroid:ktlintFormat
10:43:31.045 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "baseline" id.
10:43:31.047 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "checkstyle" id.
10:43:31.048 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "json" id.
10:43:31.048 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "format" id.
10:43:31.048 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "html" id.
10:43:31.048 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "plain" id.
10:43:31.048 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "sarif" id.
10:43:31.049 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Initializing "plain" reporter with {verbose=true, color=false, color_name=DARK_GRAY, format=true}
10:43:31.111 [main] DEBUG com.pinterest.ktlint.internal.FileUtils - Start walkFileTree for rootDir: '/Users/user/Developer/MedicineOne/androidprojectAndroid/androidprojectandroid'
   include:
[      - sun.nio.fs.UnixFileSystem$3@1139b2f3]
   exlcude:
[      - sun.nio.fs.UnixFileSystem$3@7a69b07]
10:43:32.055 [main] DEBUG com.pinterest.ktlint.internal.FileUtils - Discovered 627 files to be processed in 944 ms

> Task :androidprojectliteandroid:ktlintFormat FAILED
10:43:31.045 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "baseline" id.
10:43:31.048 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "checkstyle" id.
10:43:31.048 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "json" id.
10:43:31.048 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "format" id.
10:43:31.048 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "html" id.
10:43:31.048 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "plain" id.
10:43:31.048 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Discovered reporter with "sarif" id.
10:43:31.049 [main] DEBUG com.pinterest.ktlint.internal.KtlintCommandLine - Initializing "plain" reporter with {verbose=true, color=false, color_name=DARK_GRAY, format=true}
10:43:31.110 [main] DEBUG com.pinterest.ktlint.internal.FileUtils - Start walkFileTree for rootDir: '/Users/user/Developer/MedicineOne/androidprojectAndroid/androidprojectliteandroid'
   include:
[      - sun.nio.fs.UnixFileSystem$3@1139b2f3]
   exlcude:
[      - sun.nio.fs.UnixFileSystem$3@7a69b07]
10:43:31.451 [main] DEBUG com.pinterest.ktlint.internal.FileUtils - Discovered 25 files to be processed in 339 ms
[DEBUG] Rule with id 'standard:indent' should run after the rule with id 'experimental:function-signature'. However, the latter rule is not loaded and is allowed to be ignored. For best results, it is advised load the rule.
[DEBUG] Rules will be executed in order below (unless disabled):
           - standard:annotation, 
           - standard:annotation-spacing, 
           - standard:argument-list-wrapping, 
           - standard:chain-wrapping, 
           - standard:colon-spacing, 
           - standard:comma-spacing, 
           - standard:comment-spacing, 
           - standard:curly-spacing, 
           - standard:dot-spacing, 
           - standard:double-colon-spacing, 
           - standard:enum-entry-name-case, 
           - standard:filename, 
           - standard:final-newline, 
           - standard:import-ordering, 
           - standard:keyword-spacing, 
           - standard:modifier-order, 
           - standard:multiline-if-else, 
           - standard:no-blank-line-before-rbrace, 
           - standard:no-blank-lines-in-chained-method-calls, 
           - standard:no-consecutive-blank-lines, 
           - standard:no-empty-class-body, 
           - standard:no-empty-first-line-in-method-block, 
           - standard:no-line-break-after-else, 
           - standard:no-line-break-before-assignment, 
           - standard:no-multi-spaces, 
           - standard:no-semi, 
           - standard:no-trailing-spaces, 
           - standard:no-unit-return, 
           - standard:no-unused-imports, 
           - standard:no-wildcard-imports, 
           - standard:op-spacing, 
           - standard:package-name, 
           - standard:parameter-list-wrapping, 
           - standard:paren-spacing, 
           - standard:range-spacing, 
           - standard:spacing-around-angle-brackets, 
           - standard:spacing-between-declarations-with-annotations, 
           - standard:spacing-between-declarations-with-comments, 
           - standard:string-template, 
           - standard:unary-op-spacing, 
           - standard:wrapping, 
           - standard:indent, 
           - standard:trailing-comma-on-call-site, 
           - standard:max-line-length, 
           - standard:trailing-comma-on-declaration-site
Exception in thread "main" java.util.concurrent.ExecutionException: java.lang.ClassCastException: class java.lang.String cannot be cast to class java.lang.Boolean (java.lang.String and java.lang.Boolean are in module java.base of loader 'bootstrap')
        at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
        at com.pinterest.ktlint.internal.KtlintCommandLine.parallel(KtlintCommandLine.kt:605)
        at com.pinterest.ktlint.internal.KtlintCommandLine.parallel$default(KtlintCommandLine.kt:563)
        at com.pinterest.ktlint.internal.KtlintCommandLine.lintFiles(KtlintCommandLine.kt:321)
        at com.pinterest.ktlint.internal.KtlintCommandLine.run(KtlintCommandLine.kt:268)
        at com.pinterest.ktlint.Main.main(Main.kt:27)
Caused by: java.lang.ClassCastException: class java.lang.String cannot be cast to class java.lang.Boolean (java.lang.String and java.lang.Boolean are in module java.base of loader 'bootstrap')
        at com.pinterest.ktlint.ruleset.standard.TrailingCommaOnCallSiteRule.beforeFirstNode(TrailingCommaOnCallSiteRule.kt:56)
        at com.pinterest.ktlint.core.KtLint.executeRule(KtLint.kt:194)
        at com.pinterest.ktlint.core.KtLint.access$executeRule(KtLint.kt:30)
        at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:252)
        at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:251)
        at com.pinterest.ktlint.core.internal.VisitorProvider$visitor$4.invoke(VisitorProvider.kt:82)
        at com.pinterest.ktlint.core.internal.VisitorProvider$visitor$4.invoke(VisitorProvider.kt:80)
        at com.pinterest.ktlint.core.KtLint.format(KtLint.kt:251)
        at com.pinterest.ktlint.internal.FileUtilsKt.formatFile(FileUtils.kt:254)
        at com.pinterest.ktlint.internal.KtlintCommandLine.process(KtlintCommandLine.kt:379)
        at com.pinterest.ktlint.internal.KtlintCommandLine.access$process(KtlintCommandLine.kt:52)
        at com.pinterest.ktlint.internal.KtlintCommandLine$lintFiles$3.invoke$lambda-0(KtlintCommandLine.kt:312)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)

> Task :androidprojectandroid:ktlintFormat FAILED
[DEBUG] Rule with id 'standard:indent' should run after the rule with id 'experimental:function-signature'. However, the latter rule is not loaded and is allowed to be ignored. For best results, it is advised load the rule.
[DEBUG] Rules will be executed in order below (unless disabled):
           - standard:annotation, 
           - standard:annotation-spacing, 
           - standard:argument-list-wrapping, 
           - standard:chain-wrapping, 
           - standard:colon-spacing, 
           - standard:comma-spacing, 
           - standard:comment-spacing, 
           - standard:curly-spacing, 
           - standard:dot-spacing, 
           - standard:double-colon-spacing, 
           - standard:enum-entry-name-case, 
           - standard:filename, 
           - standard:final-newline, 
           - standard:import-ordering, 
           - standard:keyword-spacing, 
           - standard:modifier-order, 
           - standard:multiline-if-else, 
           - standard:no-blank-line-before-rbrace, 
           - standard:no-blank-lines-in-chained-method-calls, 
           - standard:no-consecutive-blank-lines, 
           - standard:no-empty-class-body, 
           - standard:no-empty-first-line-in-method-block, 
           - standard:no-line-break-after-else, 
           - standard:no-line-break-before-assignment, 
           - standard:no-multi-spaces, 
           - standard:no-semi, 
           - standard:no-trailing-spaces, 
           - standard:no-unit-return, 
           - standard:no-unused-imports, 
           - standard:no-wildcard-imports, 
           - standard:op-spacing, 
           - standard:package-name, 
           - standard:parameter-list-wrapping, 
           - standard:paren-spacing, 
           - standard:range-spacing, 
           - standard:spacing-around-angle-brackets, 
           - standard:spacing-between-declarations-with-annotations, 
           - standard:spacing-between-declarations-with-comments, 
           - standard:string-template, 
           - standard:unary-op-spacing, 
           - standard:wrapping, 
           - standard:indent, 
           - standard:trailing-comma-on-call-site, 
           - standard:max-line-length, 
           - standard:trailing-comma-on-declaration-site
Exception in thread "main" java.util.concurrent.ExecutionException: java.lang.ClassCastException: class java.lang.String cannot be cast to class java.lang.Boolean (java.lang.String and java.lang.Boolean are in module java.base of loader 'bootstrap')
        at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
        at com.pinterest.ktlint.internal.KtlintCommandLine.parallel(KtlintCommandLine.kt:605)
        at com.pinterest.ktlint.internal.KtlintCommandLine.parallel$default(KtlintCommandLine.kt:563)
        at com.pinterest.ktlint.internal.KtlintCommandLine.lintFiles(KtlintCommandLine.kt:321)
        at com.pinterest.ktlint.internal.KtlintCommandLine.run(KtlintCommandLine.kt:268)
        at com.pinterest.ktlint.Main.main(Main.kt:27)
Caused by: java.lang.ClassCastException: class java.lang.String cannot be cast to class java.lang.Boolean (java.lang.String and java.lang.Boolean are in module java.base of loader 'bootstrap')
        at com.pinterest.ktlint.ruleset.standard.TrailingCommaOnCallSiteRule.beforeFirstNode(TrailingCommaOnCallSiteRule.kt:56)
        at com.pinterest.ktlint.core.KtLint.executeRule(KtLint.kt:194)
        at com.pinterest.ktlint.core.KtLint.access$executeRule(KtLint.kt:30)
        at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:252)
        at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:251)
        at com.pinterest.ktlint.core.internal.VisitorProvider$visitor$4.invoke(VisitorProvider.kt:82)
        at com.pinterest.ktlint.core.internal.VisitorProvider$visitor$4.invoke(VisitorProvider.kt:80)
        at com.pinterest.ktlint.core.KtLint.format(KtLint.kt:251)
        at com.pinterest.ktlint.internal.FileUtilsKt.formatFile(FileUtils.kt:254)
        at com.pinterest.ktlint.internal.KtlintCommandLine.process(KtlintCommandLine.kt:379)
        at com.pinterest.ktlint.internal.KtlintCommandLine.access$process(KtlintCommandLine.kt:52)
        at com.pinterest.ktlint.internal.KtlintCommandLine$lintFiles$3.invoke$lambda-0(KtlintCommandLine.kt:312)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':androidprojectliteandroid:ktlintFormat'.
> Process 'command '/Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':androidprojectandroid:ktlintFormat'.
> Process 'command '/Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 8s
2 actionable tasks: 2 executed
```

## Steps to Reproduce
Use the following `.editorconfig` file.
```
# http://editorconfig.org
root = true

[*.{kt,kts}]
disabled_rules = filename

ij_kotlin_allow_trailing_comma = true
ij_kotlin_allow_trailing_comma_on_call_site = true
```

Run the following command on the terminal: `./gradlew ktlintFormat --daemon`

## Your Environment
* Version of ktlint used: 0.47.0
* Relevant parts of the `.editorconfig` settings
```
# http://editorconfig.org
root = true

[*.{kt,kts}]
disabled_rules = filename

ij_kotlin_allow_trailing_comma = true
ij_kotlin_allow_trailing_comma_on_call_site = true
```

* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): Gradle plugin
* Version of Gradle used (if applicable): 7.5
* Operating System and version: macOS 12.5.1

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@ This project adheres to [Semantic Versioning](https://semver.org/).
 
 ### Fixed
 * Do not add trailing comma in empty parameter/argument list with comments (`trailing-comma-on-call-site`, `trailing-comma-on-declaration-site`) ([#1602](https://github.com/pinterest/ktlint/issue/1602))
+* Fix class cast exception when specifying a non-string editorconfig setting in the default ".editorconfig" ([#1627](https://github.com/pinterest/ktlint/issue/1627)) 
 
 ### Added
 

--- a/ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/api/UsesEditorConfigProperties.kt
+++ b/ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/api/UsesEditorConfigProperties.kt
@@ -108,7 +108,25 @@ public interface UsesEditorConfigProperties {
                 }
         }
 
-        return property?.getValueAs()
+        val propertyValue =
+            when {
+                property == null -> null
+                property.type != null -> property.getValueAs()
+                else -> {
+                    // In case the property was loaded from the default ".editorconfig" the type field is not known as
+                    // the property could not yet be linked to a property type that is defined in a rule. To prevent
+                    // class cast exceptions, lookup the property by name and convert to property type.
+                    @Suppress("UNCHECKED_CAST")
+                    this@UsesEditorConfigProperties
+                        .editorConfigProperties
+                        .find { it.type.name == property.name }
+                        ?.type
+                        ?.parse(property.sourceValue)
+                        ?.parsed as T?
+                }
+            }
+
+        return propertyValue
             ?: editorConfigProperty
                 .getDefaultValue(codeStyleValue)
                 .also {

--- a/ktlint/src/test/kotlin/com/pinterest/ktlint/EditorConfigDefaultsLoaderCLITest.kt
+++ b/ktlint/src/test/kotlin/com/pinterest/ktlint/EditorConfigDefaultsLoaderCLITest.kt
@@ -95,6 +95,20 @@ class EditorConfigDefaultsLoaderCLITest : BaseCLITest() {
         }
     }
 
+    @Test
+    fun `Issue 1627 - Given a default editorconfig containing a boolean setting then do not throw a class cast exception`() {
+        val projectDirectory = "$BASE_DIR_PLACEHOLDER/editorconfig-path/project"
+        runKtLintCliProcess(
+            "editorconfig-path",
+            listOf("--editorconfig=$projectDirectory/editorconfig-boolean-setting", "--debug"),
+        ) {
+            assertErrorExitCode()
+
+            val assertThat = assertThat(errorOutput)
+            assertThat.doesNotContainLineMatching(Regex(".*java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Boolean.*"))
+        }
+    }
+
     private fun ListAssert<String>.containsLineMatching(regex: Regex) =
         this.anyMatch {
             it.matches(regex)

--- a/ktlint/src/test/resources/cli/editorconfig-path/project/editorconfig-boolean-setting
+++ b/ktlint/src/test/resources/cli/editorconfig-path/project/editorconfig-boolean-setting
@@ -1,0 +1,5 @@
+root = true
+
+[*.{kt,kts}]
+ij_kotlin_allow_trailing_comma=true
+ij_kotlin_allow_trailing_comma_on_call_site=true

